validation-test/Sema/type_checker_crashers_fixed/overloaded_generic.swift

```diff
@@ -0,0 +1,13 @@
+// RUN: not %target-swift-frontend %s -typecheck
+
+protocol A {}
+protocol B {}
+
+struct X : A {}
+struct Y : B {}
+
+struct G<T:A> { var v : T }
+struct G<T:B> { var v : T }
+
+let a = G(v: X())
+let b = G(v: Y())
```

Comment:
Updated assert makes sense, thank you! But the example itself doesn't trigger updated code in requirement failure, or any diagnostics for that matter. Would be great to get this example to diagnose before merging. I think that might be a bit tough though because fixing requirements in overloaded decls like that usually leads to ambiguities because both `G`s would much.

Reply:
Hi @xedin – If you look at the test case, it negates the exit value via `not` because the actual diagnostics doesn't matter, just that the test case should no longer crash the compiler.

Reply:
Sorry what I mean is example as-is type-checks correctly, it's not going to generate any fixes, and subsequently not going to trigger `getDeclRef()` code and crash.

Reply:
The test case does not type check as is. Before this pull request:
```
/tmp/gt.swift:8:8: error: invalid redeclaration of 'G'
struct G<T:B> { var v : T }
       ^
/tmp/gt.swift:7:8: note: 'G' previously declared here
struct G<T:A> { var v : T }
       ^
/tmp/gt.swift:11:9: error: generic struct 'G' requires that 'Y' conform to 'A'
let b = G(v: Y())
        ^
/tmp/gt.swift:7:8: note: where 'T' = 'Y'
struct G<T:A> { var v : T }
       ^
swift: /home/dave/s/u/swift/lib/Sema/CSDiagnostics.cpp:149: swift::ValueDecl *swift::constraints::RequirementFailure::getDeclRef() const: Assertion `isa<TypeExpr>(AE->getFn())' failed.
```

Reply:
Ah, sorry for the confusion, I've tried in REPL and it works differently there, would create two decls in different modules, that's why it type-checked just fine...